internal/indexer/definitions/ptp.yaml

```
@@ -93,31 +64,53 @@ irc:
     type: single
     lines:
       - tests:
-        - line: That Movie [1972] by Some Director | x264 / Blu-ray / MKV / 1080p | 204371 | 964303 | That.Movie.1972.1080p.BluRay.FLAC.x264-GROUP | comedy, drama, romance
+        - line: 1:1:357805:1332009:9k6p:tt24249072:1899875175|H.264/MKV/WEB/720p|2023|Last Straw|Last.Straw.2023.REPACK.720p.AMZN.WEB-DL.DDP5.1.H.264-FLUX|thriller,horror
           expect:
-            year: "1972"
-            releaseTags: x264 / Blu-ray / MKV / 1080p
+            new: 1
+            category: 1
+            groupId: "357805"
+            torrentId: "1332009"
+            key: "9k6p"
+            imdb: "tt24249072"
+            torrentSize: "1899875175"
+            releaseTags: H.264/MKV/WEB/720p
             freeleech: ""
-            torrentId: "964303"
-            torrentName: That.Movie.1972.1080p.BluRay.FLAC.x264-GROUP
-            tags: comedy, drama, romance
-        - line: That Other Movie [1972] | x264 / Blu-ray / MKV / 1080p / Freeleech! | 204371 | 964303 | That.Other.Movie.1972.1080p.BluRay.FLAC.x264-GROUP | comedy, drama, romance
+            year: "2023"
+            title: "Last Straw"
+            torrentName: Last.Straw.2023.REPACK.720p.AMZN.WEB-DL.DDP5.1.H.264-FLUX
+            tags: thriller,horror
+        - line: 1:1:357805:1332009:9k6p:tt24249072:1899875175|H.264/MKV/WEB/720p/FL|2023|Last Straw|Last.Straw.2023.REPACK.720p.AMZN.WEB-DL.DDP5.1.H.264-FLUX|thriller,horror
           expect:
-            year: "1972"
-            releaseTags: x264 / Blu-ray / MKV / 1080p / Freeleech!
-            freeleech: Freeleech
-            torrentId: "964303"
-            torrentName: That.Other.Movie.1972.1080p.BluRay.FLAC.x264-GROUP
-            tags: comedy, drama, romance
-        pattern: '.* \[(.*)\] (?:by .*)?\| (.*?(?: \/ (Freeleech)!)?) \| .* \| (.*) \| (.*) \| (.*)'
+            new: 1
+            category: 1
+            groupId: "357805"
+            torrentId: "1332009"
+            key: "9k6p"
+            imdb: "tt24249072"
+            torrentSize: "1899875175"
+            releaseTags: H.264/MKV/WEB/720p/FL
+            freeleech: "FL"
+            year: "2023"
+            title: "Last Straw"
+            torrentName: Last.Straw.2023.REPACK.720p.AMZN.WEB-DL.DDP5.1.H.264-FLUX
+            tags: thriller,horror
+        # <boolnew>:<category>:<gid>:<tid>:<key>:tt<imdbid]>:<sizebytes>|<format>/<container>/<source>/<resolution>(/S)?(/(FL/HL/NL)?|<year>|<title>|<releasename>|<taglist>
+        pattern: '(\d):(\d):(\d+):(\d+):(.+):(tt?\d+):(\d+)\|(.*?(?:\/(FL|HL|NL))*?)\|(\d+)\|(.+)\|(.+)\|(.+)'
```

```suggestion
        pattern: '(\d):(\d):(\d*):(\d*):(.*):(tt\d*):(\d*)\|(.*?)\/?(FL|HL|NL)?\|(\d*)\|(.*)\|(.*)\|(.*)'
```
I would suggest to replace all the + quantifiers with * in case on of the var happens to be missing.
Furthermore i think there is no IMDB ID that doesn't begin with tt so making the second t optional is not needed here?
The non-capturing group isn't needed aswell here afaict so i got rid of the nesting/non-capturing group.
Feel free to point out in case i have a mistake in my logic!

---

those changes look good, but having the * quantifier for the IMDb ID probably doesn't make sense, because if there is no id it won't match either way

---

Good point! Thanks nuxencs!
Edited the group optional in my suggestion!

The new announce line is already being used in their channel and ze0s any myself are gathering announces.

---

Ahh it's actually announcing twice, once with the old and once with the new format

---

looking at a few announnces, we could probably restrict the `key` matching to alphanumeric with `\w*`

---

Since key *should* also be there i replaced it with `\w+`

---

I'm guessing the same could be said for `gid` and `tid`? 🤔

---

`tid` yes, `gid` i am actually not sure but probably yes...
However having * here makes it more failsafe imo.
Open to discussion here though in case i didn't think about something here!

Having + for the key is actually beneficial since it seems like we need it for URL building now.